astroid/brain/brain_typing.py

```diff
@@ -370,11 +370,18 @@ def infer_typing_cast(
     node: Call, ctx: context.InferenceContext = None
 ) -> typing.Iterator[NodeNG]:
     """Infer call to cast() returning same type as casted-from var"""
+    if not isinstance(node.func, (Name, Attribute)):
+        raise UseInferenceDefault
```

Comment:
Technically this is covered by `_looks_like_typing_cast` but type checker don't know about that.